internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status.go

```diff
@@ -129,7 +129,7 @@ func NewClusterMachineConfigStatusController() *ClusterMachineConfigStatusContro
 					machineStatus.TypedSpec().Value.Schematic.Id != expectedSchematic
 
 				// don't run the upgrade check if the running version and expected versions match
-				if versionMismatch && talosVersion.TypedSpec().Value.TalosVersion != "" && talosVersion.TypedSpec().Value.SchematicId != "" {
+				if versionMismatch && talosVersion.TypedSpec().Value.TalosVersion != "" {
```

Comment:
how does it work with "invalid" schematics? is it going to wipe it?

Reply:
No, it's going to keep using the vanilla installer.
This code is in place, but this check made it skip the upgrade function as a whole.